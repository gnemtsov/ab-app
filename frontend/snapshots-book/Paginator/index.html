
            <!doctype html>
            <html lang="en">
                <head>
                    <meta charset="utf-8">
                    <title>Paginator</title>
                    <style>
            .TOCContainer a{
                display:block;
                font-size: 1.2rem; 
                margin-left: 1rem;
            }

            .SubHeader{ padding: 0.6rem; background-color: #d1d1d1; }
            .Filters span { margin-left: 1rem; }

            .TestResultContainer{ margin-top: 1.5rem; }
            .TestResultContainer.passed .TestHeader{ padding: 0.6rem; background-color: #99e99b; }
            .TestResultContainer.failed .TestHeader{ padding: 0.6rem; background-color: #ff9f9f; }
            .TestResult{
                display: flex;
                justify-content: space-around;                
            }

            .ExpectedContainer, .ActualContainer {
                padding: 0.6rem;
                width: 100%;
            }
            .ExpectedHeader, .ActualHeader { color: #4c4c4c; }
            .ExpectedContainer iframe, .ActualContainer iframe{
                width: 100%;
                height: 350px;
                border: 1px solid #beb6b6;
            }
        </style>
                    <script>
            document.addEventListener("DOMContentLoaded", function(){
                document.getElementById('show-all').addEventListener("click", function(e) {
                    e.preventDefault();
                    var resultContainers = document.getElementsByClassName("TestResultContainer");
                    for (var i = 0; i < resultContainers.length; ++i) {
                        var el = resultContainers[i];
                        el.style.display = 'block';
                    }                    
                });

                document.getElementById('show-passed').addEventListener("click", function(e) {
                    e.preventDefault();
                    var resultContainers = document.getElementsByClassName("TestResultContainer");
                    for (var i = 0; i < resultContainers.length; ++i) {
                        var el = resultContainers[i];
                        if(el.classList.contains('passed')){
                            el.style.display = 'block';
                        } else {
                            el.style.display = 'none';
                        }
                    }                    
                });

                document.getElementById('show-failed').addEventListener("click", function(e) {
                    e.preventDefault();
                    var resultContainers = document.getElementsByClassName("TestResultContainer");
                    for (var i = 0; i < resultContainers.length; ++i) {
                        var el = resultContainers[i];
                        if(el.classList.contains('failed')){
                            el.style.display = 'block';
                        } else {
                            el.style.display = 'none';
                        }
                    };
                });               
            });
        </script>
                </head>
                <body>
                <div><a href="../index.html">Table of contents</a></div>
                <h1>Paginator</h1>
                <div class="Filters">
                    Show 
                    <span><a id="show-all" href="#">all</a>(15)</span>
                    <span><a id="show-passed" href="#">passed</a>(15)</span>
                    <span><a id="show-failed" href="#">failed</a>(0)</span>
                </div>
                
                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 0, Current = 0</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="1/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 1, Current = -1</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="2/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 1, Current = 1</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="3/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 2, Current = 2</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="4/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 3, Current = 1</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="5/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 1</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="6/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 2</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="7/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 3</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="8/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 4</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="9/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 5</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="10/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 6</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="11/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 7</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="12/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 8</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="13/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 9</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="14/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                

                    <div class="TestResultContainer passed">
                        <div class="TestHeader">Total = 10, Current = 10</div>
                        <div class="TestResult">
                            
                    <div class="ExpectedContainer">
                        <div class="ExpectedHeader"></div>
                        <iframe src="15/expectedHtml.html">
                            Browser should support iframes.
                        </iframe>
                    </div>
                
                            
                        </div>
                    </div>
                
            </body>
            </html>
        