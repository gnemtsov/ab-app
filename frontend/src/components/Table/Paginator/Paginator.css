.Paginator{
    display: flex;
}

.Paginator button{
    margin: 0 0.05rem;
}

.Paginator .Dots:disabled{
    background-color: #cccccc;
    color: #333333;
}

.Paginator button:first-child{
    margin-left: 0;
}

.Paginator button:last-child{
    margin-right: 0;
}