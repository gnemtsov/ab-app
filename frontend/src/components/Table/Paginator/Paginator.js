import React from 'react';

import classes from './Paginator.css';
import Button from '../../UI/Button/Button';

const Paginator = (props) => {
    const { tp, cp, pageClickHandler } = props;
    let paginator = null;

    if (tp !== undefined && tp > 0) {
        let buttons = [];
        buttons.push(
            <Button
                key={`pback`}
                disabled={cp === 1}
                clicked={(cp === 1 ? null : event => pageClickHandler(event, 'back'))}>
                &larr;
                </Button>
        );

        const isDots = (i, tp, cp) =>
            i > 1 &&
            i < tp &&
            (i > cp + 1 || i < cp - 1) &&
            (cp > 4 || i > 5) &&
            (cp < tp - 3 || i < tp - 4);
        let flag;
        for (let i = 1; i <= tp; i++) {
            const dots = isDots(i, tp, cp) && (isDots(i - 1, tp, cp) || isDots(i + 1, tp, cp));
            if (flag && dots) {
                flag = false;
                buttons.push(
                    <Button
                        key={`p${i}`}
                        className={classes.Dots}
                        disabled={true}>
                        ...
                </Button>
                );
            } else if (!dots) {
                flag = true;
                buttons.push(
                    <Button
                        key={`p${i}`}
                        disabled={i === cp}
                        clicked={(i === cp ? null : event => pageClickHandler(event, i))}>
                        {i}
                    </Button>
                );

            }
        }

        buttons.push(
            <Button
                key={`pforward`}
                disabled={cp === tp}
                clicked={(cp === tp ? null : event => pageClickHandler(event, 'forward'))}>
                &rarr;
                </Button>
        );
        paginator =
            <div className={classes.Paginator}>
                {buttons}
            </div>
    }

    return paginator;
}

export default Paginator;