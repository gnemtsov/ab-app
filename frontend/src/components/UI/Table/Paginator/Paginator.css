.Active{
    font-weight: bold;
}
.Disabled{

}