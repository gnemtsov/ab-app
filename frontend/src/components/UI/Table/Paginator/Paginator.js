import React from 'react';

import classes from './Paginator.css';
import Button from '../../Form/Button/Button';

const Paginator = (props) => {
    let paginator = [];

    const tp = props.totalPages;
    const cp = props.currentPage;

    paginator.push(
        <Button
            key={`pback`}
            disabled={cp === 1}
            clicked={(cp === 1 ? null : event => props.pageClickHandler(event, 'back'))}>
            &larr;
                </Button>
    );

    let flag;
    for (let i = 1; i <= tp; i++) {
        const dots = i > 1 && i < tp && (i > cp + 1 || i < cp - 1) && (cp > 4 || i > 5) && (cp < tp - 3 || i < tp - 4);
        if (flag && dots) {
            flag = false;
            paginator.push(
                <button
                    key={`p${i}`}
                    disabled={true}>
                    ...
                </button>
            );
        } else if (!dots) {
            flag = true;
            paginator.push(
                <Button
                    key={`p${i}`}
                    disabled={i === cp}
                    clicked={(i === cp ? null : event => props.pageClickHandler(event, i))}>
                    {i}
                </Button>
            );

        }
    }

    paginator.push(
        <Button
            key={`pforward`}
            disabled={cp === tp}
            clicked={(cp === tp ? null : event => props.pageClickHandler(event, 'forward'))}>
            &rarr;
                </Button>
    );

    return (
        <div className={classes.Paginator}>
            {paginator}
        </div>
    );
}

export default Paginator;