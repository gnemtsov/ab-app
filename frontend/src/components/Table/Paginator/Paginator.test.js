import React from 'react';

import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Paginator from './Paginator';

configure({ adapter: new Adapter() });

describe('Paginator', () => {
    const snapshoot = (tp, cp) => {
        const wrapper = render(<Paginator tp={tp} cp={cp} />);
        it(`Total = ${tp}, Current = ${cp}`, () => {
            expect(wrapper).toMatchSnapshot();
        });
    }

    snapshoot(0, 0);
    snapshoot(1, -1);
    snapshoot(1, 1);
    snapshoot(2, 2);
    snapshoot(3, 1);

    for (let cp = 1; cp <= 10; cp++) {
        snapshoot(10, cp);
    }
});
