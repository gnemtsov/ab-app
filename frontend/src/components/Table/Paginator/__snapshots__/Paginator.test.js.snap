// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Total = 0, Current = 0 1`] = `null`;

exports[`Paginator Total = 1, Current = -1 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 1, Current = 1 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
    disabled=""
  >
    ←
  </button>
  <button
    class="Button"
    disabled=""
  >
    1
  </button>
  <button
    class="Button"
    disabled=""
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 2, Current = 2 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button"
    disabled=""
  >
    2
  </button>
  <button
    class="Button"
    disabled=""
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 3, Current = 1 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
    disabled=""
  >
    ←
  </button>
  <button
    class="Button"
    disabled=""
  >
    1
  </button>
  <button
    class="Button"
  >
    2
  </button>
  <button
    class="Button"
  >
    3
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 1 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
    disabled=""
  >
    ←
  </button>
  <button
    class="Button"
    disabled=""
  >
    1
  </button>
  <button
    class="Button"
  >
    2
  </button>
  <button
    class="Button"
  >
    3
  </button>
  <button
    class="Button"
  >
    4
  </button>
  <button
    class="Button"
  >
    5
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 2 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button"
    disabled=""
  >
    2
  </button>
  <button
    class="Button"
  >
    3
  </button>
  <button
    class="Button"
  >
    4
  </button>
  <button
    class="Button"
  >
    5
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 3 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button"
  >
    2
  </button>
  <button
    class="Button"
    disabled=""
  >
    3
  </button>
  <button
    class="Button"
  >
    4
  </button>
  <button
    class="Button"
  >
    5
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 4 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button"
  >
    2
  </button>
  <button
    class="Button"
  >
    3
  </button>
  <button
    class="Button"
    disabled=""
  >
    4
  </button>
  <button
    class="Button"
  >
    5
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 5 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    4
  </button>
  <button
    class="Button"
    disabled=""
  >
    5
  </button>
  <button
    class="Button"
  >
    6
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 6 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    5
  </button>
  <button
    class="Button"
    disabled=""
  >
    6
  </button>
  <button
    class="Button"
  >
    7
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 7 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    6
  </button>
  <button
    class="Button"
    disabled=""
  >
    7
  </button>
  <button
    class="Button"
  >
    8
  </button>
  <button
    class="Button"
  >
    9
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 8 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    6
  </button>
  <button
    class="Button"
  >
    7
  </button>
  <button
    class="Button"
    disabled=""
  >
    8
  </button>
  <button
    class="Button"
  >
    9
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 9 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    6
  </button>
  <button
    class="Button"
  >
    7
  </button>
  <button
    class="Button"
  >
    8
  </button>
  <button
    class="Button"
    disabled=""
  >
    9
  </button>
  <button
    class="Button"
  >
    10
  </button>
  <button
    class="Button"
  >
    →
  </button>
</div>
`;

exports[`Paginator Total = 10, Current = 10 1`] = `
<div
  class="Paginator"
>
  <button
    class="Button"
  >
    ←
  </button>
  <button
    class="Button"
  >
    1
  </button>
  <button
    class="Button Dots"
    disabled=""
  >
    ...
  </button>
  <button
    class="Button"
  >
    6
  </button>
  <button
    class="Button"
  >
    7
  </button>
  <button
    class="Button"
  >
    8
  </button>
  <button
    class="Button"
  >
    9
  </button>
  <button
    class="Button"
    disabled=""
  >
    10
  </button>
  <button
    class="Button"
    disabled=""
  >
    →
  </button>
</div>
`;
